### PR TITLE
Feat: support performance budget

### PR DESCRIPTION
**Pull request by farhan-sauce**
### support performance budget
Issue: https://github.com/saucelabs/speedo/issues/71

`budget` can be set in `speedo.config.js` file to strictly assert metrics against budget using `analyze` or `run` command.

example: 
```js
// speedo.config.js
module.exports = {
    budget: {
        load: 500, // maximum
        speedIndex: [100, 500] // [minium, maximum]
    }
}
```

### CODE REVIEW

**Comment by farhan-sauce** [comment]
support budget in `run` command is pending. I will reopen once that will be done.

File: README.md [hunk 1]
```diff
@@ -155,6 +155,23 @@ Define the identifier for Sauce Connect tunnel to run performance tests for loca
 
 Define the username of your parent account that is running Sauce Connect tunnel.
 
+##### `performanceBudget`
```
Comment (christian-bromann) [inline README.md]: let's just call it `budget`

File: tests/__snapshots__/analyze.test.js.snap [hunk 2]
```diff
@@ -4,636 +4,7 @@ exports[`should fail if asserting second order index 1`] = `
 Array [
   Array [
     Object {
-      "id": "9",
-      "name": undefined,
-      "passed": true,
```
Comment (farhan-sauce) [inline tests/__snapshots__/analyze.test.js.snap]: It should return `passed: false` for `should fail if asserting second order index` test.

After adding `analyzeReport.mockClear()` in `afterEach` it stores `passed: false`.

File: src/commands/run.js [hunk 3]
```diff
@@ -242,6 +243,33 @@ export const handler = async (argv) => {
...
+        })
+        const preparedBudget = prepareBudgetData(budget)
+        for (const [metric, [{ l: lowerLimit, u: upperLimit }]] of Object.entries(preparedBudget)) {
+            const actual = performanceLog[0].metrics[metric]
+            /**
+             * don't fail if:
+             * - metric is within budget
+             * - we don't care about this metric
+             */
+            if (actual <= upperLimit && actual >= lowerLimit) {
```
Comment (christian-bromann) [inline src/commands/run.js]: in this if statement we only check if the metric is within the budget but we don't have a check whether or not we care about it. Did you miss that or is this part of a different check.

File: README.md [hunk 4]
```diff
@@ -155,6 +155,23 @@ Define the identifier for Sauce Connect tunnel to run performance tests for loca
...
+// speedo.config.js
+module.exports = {
+    budget: {
+        load: 500, // maximum
+        speedIndex: [100, 500] // [minium, maximum]
+    }
+}
+```
+
+in above example if `load` metric value is greater t
```
Comment (christian-bromann) [inline README.md]: this sentence is cut off

File: src/commands/run.js [hunk 5]
```diff
@@ -28,7 +28,8 @@ export const handler = async (argv) => {
     const accessKey = config.key || process.env.SAUCE_ACCESS_KEY
     const jobName = getJobName(config)
     const buildName = config.build || `${jobName} - ${(new Date()).toString()}`
-    const metrics = getMetricParams(config)
+    const budget = config ? config.budget : null
+    const metrics = budget ? Object.keys(budget) : getMetricParams(config)
```
Comment (christian-bromann) [inline src/commands/run.js]: What happens if someone has a spelling error in the metrics? Do we handle the case properly?

Comment (farhan-sauce) [reply src/commands/run.js]: oh no, I will add validation logic.

**Pull request merged**